include/gz/math/MecanumDriveOdometry.hh

```diff
@@ -0,0 +1,134 @@
+    ///       |
+    ///       |
+    ///       O--->X(forward)
+    class IGNITION_MATH_VISIBLE MecanumDriveOdometry
+    {
+      // Use a steady clock
+      public: using clock = std::chrono::steady_clock;
+
+      /// \brief Constructor.
+      /// \param[in] _windowSize Rolling window size used to compute the
+      /// velocity mean
+      public: explicit MecanumDriveOdometry(size_t _windowSize = 10);
+
+      /// \brief Destructor.
+      public: ~MecanumDriveOdometry();
+
+      /// \brief Initialize the odometry
+      /// \param[in] _time Current time.
+      public: void Init(const clock::time_point &_time);
+
+      /// \brief Get whether Init has been called.
+      /// \return True if Init has been called, false otherwise.
+      public: bool Initialized() const;
+
+      /// \brief Updates the odometry class with latest wheels and
+      /// steerings position
+      /// \param[in] _frontLeftPos Left wheel position in radians.
+      /// \param[in] _frontRightPos Right wheel postion in radians.
+      /// \param[in] _backLeftPos Left wheel position in radians.
+      /// \param[in] _backRightPos Right wheel postion in radians.
+      /// \param[in] _time Current time point.
+      /// \return True if the odometry is actually updated.
+      public: bool Update(
+        const Angle &_frontLeftPos, const Angle &_frontRightPos,
+        const Angle &_backLeftPos, const Angle &_backRightPos,
+        const clock::time_point &_time);
+
+      /// \brief Get the heading.
+      /// \return The heading in radians.
+      public: const Angle &Heading() const;
+
+      /// \brief Get the X position.
+      ///  \return The X position in meters
+      public: double X() const;
+
+      /// \brief Get the Y position.
+      /// \return The Y position in meters.
+      public: double Y() const;
+
+      /// \brief Get the linear velocity.
+      /// \return The linear velocity in meter/second.
+      public: double LinearVelocity() const;
+
+      /// \brief Get the lateral velocity.
+      /// \return The lateral velocity in meter/second.
+      public: double LateralVelocity() const;
+
+      /// \brief Get the angular velocity.
+      /// \return The angular velocity in radian/second.
+      public: const Angle &AngularVelocity() const;
+
+      /// \brief Set the wheel parameters including the radius and separation.
+      /// \param[in] _wheelSeparation Distance between left and right wheels.
+      /// \param[in] _wheelBase Distance between front and back wheels.
+      /// \param[in] _leftWheelRadius Radius of the left wheel.
+      /// \param[in] _rightWheelRadius Radius of the right wheel.
+      public: void SetWheelParams(double _wheelSeparation, double _wheelBase,
+                      double _leftWheelRadius, double _rightWheelRadius);
+
+      /// \brief Set the velocity rolling window size.
+      /// \param[in] _size The Velocity rolling window size.
+      public: void SetVelocityRollingWindowSize(size_t _size);
```

Comment:
For a follow-up PR, we should also add getters for the parameters set by these API.

Reply:
Went ahead and did it.